liquibase-standard/src/main/java/liquibase/change/AbstractChange.java

```diff
@@ -868,4 +868,21 @@ public String getDescription() {
         return description;
     }
 
+    @Override
+    public boolean equals(Object reference) {
+        if (this == reference) return true;
+        if (reference instanceof DbmsTargetedChange && this instanceof DbmsTargetedChange &&
+            !((DbmsTargetedChange)reference).getDbms().equals(((DbmsTargetedChange)this).getDbms())) {
+            return false;
+        }
+        if (reference == null || getClass() != reference.getClass()) return false;
```

Comment:
Does this null check need to happen right after the same-object check to avoid an NPE?  The instanceof call will allow it a null.

I know that is an edge case.

Reply:
I simulated it here, but "reference instanceof DbmsTargetedChange" returns false so the second if will bail out on the first condition and won't call the getDbms methods.